lib/modules/module.rb

```diff
@@ -42,5 +42,23 @@ def self.rule(name,inputs,outputs,&block)
     def self.convert_risk_to_timestep(risk, original_period)
       return 1-((1-risk) ** (Synthea::Config.time_step.to_f/original_period))
     end
+
+    #These functions are used to start/update medications and to stop medications, respectively.
+    #Similar functions were not written for care plans because a care plan is typically only started/updated in one
+    #rule, whereas it is common for a medication to be used in multiple rules. This may change in future modules(?) 
+    def prescribeMedication(med, reason, time, entity)
+      entity[:medications] ||= Hash.new
+      if entity[:medications][med].nil?
+        entity[:medications][med] = [ time, [reason ] ]
```

Comment:
As noted previously (and discussed on slack), let's track time and reason with a hash rather than an array.